app/controllers/api/v1/reservations.rb

```diff
@@ -0,0 +1,43 @@
+class Api::V1::ReservationsController < ApplicationController
+  before_action :set_reservation, only: %i[create]
+  before_action :set_user, only: %i[update destroy]
+  before_action :authorized
+
+  def index
+    @reservations = if current_user.admin == true
+                      Reservation.all
+                    else
+                      current_user.reservations
+                    end
+    render json: { message: 'Showing Reservations', reservations: @reservations }, status: :ok
+  end
+
+  def create
+    @reservation = Reservation.new(reservation_params)
+    @reservation.user_id = current_user.id
+    @reservation.car_id = Car.find(params[:car_id])
```

Comment:
In your reservation params, you already obtain the user id and the car id from the form. These lines of code perform duplicate functionality and may cause errors. Please try to stick to only one way of setting the user and car id

Reply:
Oh your are right. Thanks @Shisui6